driver-async/src/main/com/mongodb/async/client/AbstractSubscription.java

```diff
@@ -124,7 +124,11 @@ void addToQueue(@Nullable final List<TResult> results) {
 
     void onError(final Throwable t) {
         if (terminalAction()) {
-            postTerminate();
+            try {
+                postTerminate();
+            } catch (Throwable throwable) {
+                // fall through
```

Comment:
I think we should move this try and catch into `MongoIterableSubscription#postTerminate` - especially as it can be called from multiple locations.

Reply:
This makes much more sense. I have moved the try-catch block to `MongoIterableSubscription#postTerminate`.